src/lib/score/graphics/widgets/QGraphicsRangeSlider.cpp

```diff
@@ -0,0 +1,138 @@
+#include <score/graphics/widgets/QGraphicsRangeSlider.hpp>
+#include <score/model/Skin.hpp>
+
+#include <QGraphicsSceneMouseEvent>
+#include <QPainter>
+
+#include <wobjectimpl.h>
+W_OBJECT_IMPL(score::QGraphicsRangeSlider);
+
+namespace score
+{
+QGraphicsRangeSlider::QGraphicsRangeSlider(QGraphicsItem* parent)
+{
+  auto& skin = score::Skin::instance();
+  setCursor(skin.CursorPointingHand);
+}
+
+void QGraphicsRangeSlider::setStart(double start)
+{
+  if (m_start != start)
+  {
+    m_start = std::clamp(start, 0., 1.);
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::setEnd(double end)
+{
+  if (m_end != end)
+  {
+    m_end = std::clamp(end, 0., 1.);
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::setValue(ossia::vec2f value)
+{
+  setStart(value[0]);
+  setEnd(value[1]);
+}
+
+ossia::vec2f QGraphicsRangeSlider::value() const noexcept
+{
+  return {float(m_start), float(m_end)};
+}
+
+void QGraphicsRangeSlider::setRange(double min, double max)
+{
+  {
+    m_min = min;
+    m_max = max;
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::mousePressEvent(QGraphicsSceneMouseEvent* event)
+{
+  if (event->button() == Qt::LeftButton)
+  {
+    d2s = abs(event->pos().x() - m_start * boundingRect().width());
+    d2c = abs(
+        event->pos().x()
+        - (m_start + (m_end - m_start) / 2) * boundingRect().width());
+    d2e = abs(event->pos().x() - m_end * boundingRect().width());
+
+    if (d2s < d2c && d2s < d2e)
+      handle = START;
+    else if (d2e < d2c && d2e < d2s)
+      handle = END;
+    else if (d2c < d2s && d2c < d2e)
+    {
+      handle = CENTER;
+      ypos = event->pos().y();
+    }
+    else
+      handle = NONE;
+  }
+}
+
+void QGraphicsRangeSlider::mouseMoveEvent(QGraphicsSceneMouseEvent* event)
+{
+  switch (handle)
+  {
```

Comment:
Same than above (and since it's only the width which is used, it 's even better and more readable to store it at the beginning